### PR TITLE
Add cardinality control for groupby benchs with flat types

### LINKED ISSUE #15114: [FEA] Add cardinality axis to nvbench for groupby aggregations
**Is your feature request related to a problem? Please describe.**
Currently the groupby benchmarks default to maximum cardinality, but it would be useful to sweep through or monitor low cardinality values where we expect to see lower throughput.

**Describe the solution you'd like**
We should add an integer cardinality axis to the groupby nvbenchmarks. It could default to `.add_int64_axis("cardinality", {0});` or perhaps `.add_int64_axis("cardinality", {0, 1000});` to align with the use of cardinality in the cuIO benchmarks.

### PR DESCRIPTION
**Pull request by PointKernel**
## Description
Contributes to #15114

This PR adds cardinality control to `group_max`, `group_nunique` and `group_rank` benchmarks.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by GregoryKimball** [comment]
Thanks for kicking this off! You might also consider adding nvbench axes for cardinality for the existing benchmarks. If we started with a default value of `{0}` then this would let us vary cardinality for engineering studies without changing the automated benchmark runtime.

File: cpp/benchmarks/groupby/group_nunique.cpp [hunk 1]
```diff
@@ -39,17 +39,23 @@ auto make_aggregation_request_vector(cudf::column_view const& values, Args&&...
...
+  auto const cardinality = static_cast<cudf::size_type>(state.get_int64("cardinality"));
 
   auto const keys = [&] {
-    data_profile profile = data_profile_builder().cardinality(0).no_validity().distribution(
-      cudf::type_to_id<int32_t>(), distribution_id::UNIFORM, 0, 100);
+    data_profile profile =
+      data_profile_builder()
+        .cardinality(cardinality)
+        .no_validity()
+        .distribution(cudf::type_to_id<int32_t>(), distribution_id::UNIFORM, 0, size);
```
Comment (ttnghia) [inline cpp/benchmarks/groupby/group_nunique.cpp:50]: Code to create `keys` and `values` here seem to be very similar (the same) as for creating those of groupby max. Can we further extract them into a common function? Like `create_keys_values()`?

Comment (PointKernel) [reply cpp/benchmarks/groupby/group_nunique.cpp:50]: The duplicate part is just data profile construction. Though we repeat those 5 lines of code each time, it's just one API call. It's not worth creating a helper function wrapping one single call IMO.

**Review by ttnghia — APPROVED** [review]

**Review by vyasr — APPROVED** [review]

**Pull request merged**